### DISCUSSION TITLE
Serve handler working after I call /api/inngest but dev server and functions not recognized

### DISCUSSION THREAD

**Discussion by mikececco** [inngest/inngest#1347] [General]
As you can see calling the endpoint is working:
![image](https://github.com/inngest/inngest/assets/88579001/b0c22dc8-c24c-4af3-9874-5f601c64d239)

but when I try to sync app/detect functions they are not recognized. I get `We could not reach your URL. Please verify that your URL is correct and check your server logs.`

my api/index.ts:
`
import {generateContent} from './inngest/generateContent.js';
import { inngest } from "./inngest/client.js";
import { serve } from "inngest/express";

app.use("/api/inngest", express.json());
app.use("/api/inngest", serve({client: inngest, functions: [generateContent] as any}));`

my api/inngest/index.ts:
`export { inngest } from "./client.js";

import { generateContent } from "./generateContent.js";
import { Request, Response } from 'express';

export default async (req: Request, res: Response) => {
  try {
    // const event = { /* construct event object from req if needed */ };
    const hand = await generateContent;

    // console.log(req);
    res.json(hand);

  } catch (error) {
    console.error('Error processing image:', error);
    res.status(500).send('Error processing image');
  }
};
`

and part of my /api/inngest/generateContent.ts:
`export const generateContent = inngest.createFunction(
  { id: "generate-content" },
  { event: "generateContent" },
  async ({ event }) => {...`
  
  Why are they not recognized?

**Comment by djfarrelly** [comment]
Hey @mikececco - Your function and `serve` code look correct. I'm confused about the function including `export default async (req: Request, res: Response) ` - what is that code? It doesn't look like any Inngest code so maybe it's something else.

With that error, it seems there might be some sort of protection (e.g. Vercel deployment protection) or otherwise on your URL. 

Do you have your `INNGEST_SIGING_KEY` set? If you are using Inngest cloud, we can pull logs to try to help debug what the exact response from your server is (hello@inngest.com). 

If you already sorted it out, please let us know what the issue was!
